proof/invariant-abstract/X64/ArchInvariants_AI.thy

```diff
@@ -3048,9 +3109,25 @@ lemma vs_lookup_2ConsD:
   apply (fastforce simp: vs_lookup1_def)
   done
 
-lemma global_refs_asid_table_update [iff]:
-  "global_refs (s\<lparr>arch_state := x64_asid_table_update f (arch_state s)\<rparr>) = global_refs s"
-  by (simp add: global_refs_def)
+lemma global_refs_updates[simp]:
+  "\<And>f. global_refs (s\<lparr>arch_state := x64_asid_table_update f (arch_state s)\<rparr>) = global_refs s"
+  "\<And>f. global_refs (s\<lparr>arch_state := x64_current_fpu_owner_update f (arch_state s)\<rparr>) = global_refs s"
+  by (auto simp: global_refs_def)
+
+(* FIXME: pspace_update_eq is interpreted for arch_state_update later, in Invariants_AI *)
+lemma valid_arch_state_updates[simp]:
+  "\<And>f. valid_arch_state (s\<lparr>arch_state := x64_current_fpu_owner_update f (arch_state s)\<rparr>) = valid_arch_state s"
+  by (auto simp: valid_arch_state_def valid_asid_table_def valid_global_pts_def valid_global_pds_def
+                 valid_global_pdpts_def valid_ioports_def all_ioports_issued_def issued_ioports_def
+                 Arch_pspace_update_eq.caps_of_state_update Arch_pspace_update_eq_def pspace_update_eq_def
+                 obj_at_def a_type_simps)
+
+(* FIXME: pspace_update_eq is interpreted for arch_state_update later, in Invariants_AI *)
```

Comment:
how do you propose we fix these?

Reply:
That is a good question. The specific issue is that I'm using a lemma from `pspace_update_eq` and manually unfolding its axioms to show that they're satisfied. If we're fine with that then I could remove the FIXMEs.

I also just realised that this could probably be resolved by moving these lemmas into `pspace_update_eq`. That would have the problem though that it's then not consistent with AARCH64, which doesn't need to do that due to these properties there not being dependent on `caps_of_state`.